task/move_test.go

```diff
@@ -2,21 +2,25 @@ package main
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"github.com/google/go-cmp/cmp"
 	"reflect"
 	"testing"
 )
 
 type stub struct {
-	toFetch []Thing
-	curr    int
-	gotPut  []Thing
+	toFetch          []Thing
+	curr             int
+	fetchErr, putErr bool
```

Comment:
A more general way that you can do this is to have `fetchEmptyErr` error and then have it return the error only when `curr > len(toFetch)`. This allows you to test situations where the error isn't immediate.